test/unit_tests/workflow_def_test.go

```diff
@@ -54,3 +55,37 @@ func TestHttpTask(t *testing.T) {
 	json, _ := json.Marshal(workflowDef)
 	fmt.Println(string(json))
 }
+
+func TestUpdateTaskWithTaskId(t *testing.T) {
+
+	updateTask :=  workflow.NewUpdateTaskWithTaskId("update_task_ref", model.CompletedTask, "target_task_to_update")
+	updateTask.MergeOutput(true)
+	updateTask.TaskOutput(map[string]interface{}{"key": map[string]interface{}{"nestedKey": "nestedValue"}})
+
+	wf := workflow.NewConductorWorkflow(testdata.WorkflowExecutor).
+		Name("workflow_with_update_task").
+		Version(1).
+		Add(updateTask)
+	workflowDef := wf.ToWorkflowDef()
+	assert.NotNil(t, workflowDef)
+	assert.Equal(t, 1, len(workflowDef.Tasks))
```

Comment:
is there anything more that we could assert to be sure the Update Task is correct? 

This is just checking that there is one task, but no check on the task itself, e.g.:  is it the right type? does it have the right inputs?

Reply:
Fair point.